themes/default/content/docs/concepts/options/transforms.md

```diff
@@ -0,0 +1,329 @@
+title: "transforms"
+h1: "Resource option: transforms"
+meta_image: /images/docs/meta-images/docs-meta.png
+menu:
+  concepts:
+    identifier: transforms
+    parent: options
+    weight: 15
+aliases:
+- /docs/intro/concepts/resources/options/transforms/
+---
+
+The `transforms` resource option provides a list of transforms to apply to a resource and all of its children. This option is used to override or modify the inputs to the child resources of a component resource. One example is to use the option to add other resource options (such as `ignoreChanges` or `protect`). Another example is to modify an input property (such as adding to tags or changing a property that is not directly configurable).
+
+Each transform is a callback that gets invoked by the Pulumi engine. It receives the resource type, name, input properties, and resource options. The callback returns a new set of resource input properties and resource options that will be used to construct the resource instead of the original values.
+
+## Transforms vs Transformations
+
+The `transforms` option is the replacement system to [transformations](/docs/concepts/options/transformations/). Calling transform functions is managed by the engine, rather than by each SDK and so transforms apply to resources created inside remote component resources (such as those in [AWSX](https://www.pulumi.com/registry/packages/awsx/)).
+
+However because `transforms` are invoked by the engine the properties are the values after being serialised and deserialised across the wire protocol to the engine and back to the SDK runtime. As such unlike with `transformations` there is no `Resource` object, and the properties will be raw property maps not typed resource arg classes.
+
+The new transform system is also natively async. As such depending on the language your using you may have to lift your transform function to be `async` and/or to handle an extra parameter for the async context (e.g. a `context.Context` in Go, or `System.Threading.CancellationToken` in C#).
+
+As such when moving from `transformations` to `transforms` you will need to update your transform code to handle these differences.
+
+### No resource type
+
+There is no `Resource` object passed to transform functions. But most of the information you could have retrieved from that object is presented on the transform arguments directly.
+
+For example given the following C# transform that used the `Resource` object to look up the Pulumi type:
+
+```csharp
+args =>
+{
+    if (args.Resource.GetResourceType() == "random:index/randomString:RandomString")
+    {
+        var resultOpts = CustomResourceOptions.Merge((CustomResourceOptions)args.Options,
+            new CustomResourceOptions {AdditionalSecretOutputs = {"length"}});
+        return new ResourceTransformationResult(resultArgs, resultOpts);
+    }
+    return null;
+}
+```
```

Comment:
We’ll need these code snippets in all languages.

Reply:
This one only really makes sense for C#. The other languages don't have `GetResourceType` like this.